hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -906,14 +906,19 @@ ReplicaInfo getReplicaInfo(String bpid, long blkid)
     return info;
   }
 
+  ReplicaInfo getReplicaInfoForLock(ExtendedBlock b)
```

Comment:
Great progress here. How about to integrate `getReplicaInfoForLock` ang `getStorageUuid` together to `getStorageUuidForLock`?

```
  String getStorageUuidForLock(ExtendedBlock b)
      throws ReplicaNotFoundException {
    return getReplicaInfo(b.getBlockPoolId(), b.getBlockId()).getStorageUuid();
  }
```

Reply:
Good idea.

Reply:
Update the patch, please review again.